react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+
+
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {displayIndex: 0, displayId: this.props.array[0].id};
+    this.imageSrc = null;
+    this.newImg = 0;
+    this.currentPath = null;
+    this.newId = null;
+    this.handleClick = this.handleClick.bind(this);
+    this.circleClass = "far fa-circle m-2";
+    this.circles = null;
+    this.intervalId = null;
```

Comment:
This is a lot of properties! You can solve this without so many additional properties, you really only need one to hold the `intervalId`